tests/data/dataset_readers/nlvr_test.py

```diff
@@ -37,3 +38,17 @@ def test_agenda_indices_are_correct(self):
         # pylint: disable=protected-access
         expected_agenda_actions = reader._get_agenda_for_sentence(sentence)
         assert expected_agenda_actions == agenda_actions
+
+    def test_agenda_indices_are_correct_with_paths(self):
+        reader = NlvrDatasetReader()
+        test_file = "tests/fixtures/data/nlvr/sample_data.jsonl"
+        dataset = reader.read(test_file)
+        instance = dataset.instances[0]
+        sentence_tokens = instance.fields["sentence"].tokens
+        sentence = " ".join([t.text for t in sentence_tokens])
+        agenda = [item.sequence_index for item in instance.fields["agenda"].field_list]
+        actions = [action.rule for action in instance.fields["actions"].field_list]
+        agenda_actions = [actions[i] for i in agenda]
+        # pylint: disable=protected-access
+        expected_agenda_actions = reader._get_agenda_for_sentence(sentence, NlvrWorld({}))
+        assert expected_agenda_actions == agenda_actions
```

Comment:
I'd like to see an explicit test for `get_path_to_root` (in `world_test.py`, not this file).  You could construct a `World`, override `all_possible_actions()` to return a simple list, and make sure that `get_path_to_root` returns what you expect.

Reply:
Done.